src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -220,63 +61,59 @@ internal static bool ConvertToIntegratedSecurity(object value)
                     else if (StringComparer.OrdinalIgnoreCase.Equals(tmp, "false") || StringComparer.OrdinalIgnoreCase.Equals(tmp, "no"))
                         return false;
                 }
-                return Boolean.Parse(svalue);
+                return bool.Parse(svalue);
             }
             try
             {
-                return ((IConvertible)value).ToBoolean(CultureInfo.InvariantCulture);
+                return Convert.ToBoolean(value, CultureInfo.InvariantCulture);
             }
             catch (InvalidCastException e)
             {
-                throw ADP.ConvertFailed(value.GetType(), typeof(Boolean), e);
+                throw ADP.ConvertFailed(value.GetType(), typeof(bool), e);
             }
         }
 
         internal static int ConvertToInt32(object value)
         {
             try
             {
-                return ((IConvertible)value).ToInt32(CultureInfo.InvariantCulture);
+                return Convert.ToInt32(value, CultureInfo.InvariantCulture);
             }
             catch (InvalidCastException e)
             {
-                throw ADP.ConvertFailed(value.GetType(), typeof(Int32), e);
+                throw ADP.ConvertFailed(value.GetType(), typeof(int), e);
             }
         }
 
         internal static string ConvertToString(object value)
         {
             try
             {
-                return ((IConvertible)value).ToString(CultureInfo.InvariantCulture);
+                return Convert.ToString(value, CultureInfo.InvariantCulture);
             }
             catch (InvalidCastException e)
             {
-                throw ADP.ConvertFailed(value.GetType(), typeof(String), e);
+                throw ADP.ConvertFailed(value.GetType(), typeof(string), e);
```

Comment:
When looking at this in the diff I wondered if value was guaranteed to be not-null here. What do you think of enabling nullable reference types for some smaller files and seeing if they help us see any possible correctness problems?

Reply:
This block catches the `InvalidCastException` here, which means it couldn't be null at this point; in addition, `Conver.Toxxx()` functions we've used in this type don't raise an exception in facing with null input parameter.
Regarding your question on the second part, could you elaborate on it with more details?

Reply:
> Regarding your question on the second part, could you elaborate on it with more details?

https://devblogs.microsoft.com/dotnet/nullable-reference-types-in-csharp/ we can use that per file or even per scope to help get safer code.

Reply:
Thank you @Wraith2, by enabling it I managed to find more possible improvements in this file.